File: simulation_prototype/Person.py
from PersonInterface import PersonInterface


class Person(PersonInterface):
    def __init__(self, new_id, new_name="Stefan"):
        super().__init__()

        self._ID = new_id
        self._name = new_name


    def getID(self):
        return self._ID


    def getName(self):
        return self._name

File: Person.cpp
#include "Person.h"

Person::Person(int new_id, std::string new_name)
{
    ID = new_id;
    name = new_name;
}

int Person::getID()
{
    return ID;
}

std::string Person::getName()
{
    return name;
}

Person::~Person() 
{

}

int Person::getNumberOfTasks() { return 0; }

void Person::addTask(Task*) {}

void Person::doTask(std::vector<Person*>*, std::vector<Order*>*) {}

void Person::setState(CustomerStates) {}

CustomerStates Person::getState() { return CustomerStates::NEW; }

void Person::doAction(std::vector<Person*>*, std::vector<Table*>*) {}

void Person::setOrder(Order*) {}

void Person::setMenu(Menu*) {}
